syntax = "proto3";

import "bob_emploi/frontend/api/options.proto";

package bayes.bob;

message Association {
  // Name of the association.
  string name = 1;

  // Link to the association home page.
  string link = 2 [(string_format) = URL_FORMAT];

  // A set of filter IDs: do not show the association if one these filters
  // returns false for the project.
  repeated string filters = 3 [(string_format) = SCORING_MODEL_ID];
}

message Associations {
  // A sorted list of associations. The ones that Bayes thinks are the best
  // suited are first.
  repeated Association associations = 1;
}

message VolunteeringMission {
  // Name of the association that requested the mission.
  string association_name = 1;

  // Short title for this mission.
  string title = 2;

  // Longer description for the mission, may use markdown.
  string description = 3;

  // Link to the mission page with full info and way to apply.
  string link = 4 [(string_format) = URL_FORMAT];

  // Whether the mission can be done from anywhere.
  bool is_available_everywhere = 5;

  // Starting date of the mission.
  string start_date = 6;

  // Duration of the mission.
  string duration = 7;

  // Domain of the mission.
  string domain = 8;
}

message VolunteeringMissions {
  // List of missions sorted by relevance for the user, or project.
  repeated VolunteeringMission missions = 1;
}

// A simple link to an external page.
message SimpleLink {
  // Path part of the URL of the page.
  string path = 1;
}

message MissionLocaleData {
  // URL to a list of mission locale agencies close to the user.
  string agencies_list_link = 1;
}
